pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -55,13 +59,56 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 			continue
 		}
 		dm.UUIDandVersion.UUID = uuid
+		if !isXen {
+			// XXX from observation, the dm.CPUTotal gets from the kvm hypervisor containerd stat is
+			// hundreds percentage of the app uptime in seconds.
+			// Assume the CPUTotal value and the app CPU usage is proportional, this nonXenCPUfactor
+			// is obtained by run the same App under Xen and KVM in comparison to be used as
+			// an approcimation.
+			dm.CPUTotal = dm.CPUTotal / nonXenCPUfactor
+		}
 		ctx.pubDomainMetric.Publish(dm.Key(), dm)
 	}
 
 	hm, _ := hyper.GetHostCPUMem()
+	if !isXen {
+		// the the hypervisor other than Xen, we don't have the Dom0 stats. Get the host
+		// cpu and memory for the device here
+		formatAndPublishHostCPUMem(ctx, hm)
+	}
 	ctx.pubHostMemory.Publish("global", hm)
 }
 
+func formatAndPublishHostCPUMem(ctx *domainContext, hm types.HostMemory) {
+	cpuInfo, err := cpu.Get()
+	if err != nil {
+		log.Errorf("getAndPublishMetrics: cpu Get error %v", err)
+		return
+	}
+
+	used := hm.TotalMemoryMB - hm.FreeMemoryMB
+	var usedPerc float64
+	if hm.TotalMemoryMB > 0 {
+		usedPerc = float64(used * 100.0 / hm.TotalMemoryMB)
+	}
+	var hostUUID types.UUIDandVersion
+	hostUUID.UUID = nilUUID
+	uptime, _ := host.Uptime()
+	cpuUsage := cpuInfo.User + cpuInfo.System + cpuInfo.Iowait + cpuInfo.Softirq + cpuInfo.Irq
```

Comment:
Iowait is not CPU time you use. That's just time spent sleeping while something else can use the CPU.

Reply:
ok.